src/agentscope_runtime/sandbox/manager/sandbox_manager.py

```diff
@@ -493,27 +500,56 @@ def create(
                 )
                 return None
 
-            # TODO: update ContainerModel according to images & backend
-            container_model = ContainerModel(
-                session_id=session_id,
-                container_id=_id,
-                container_name=container_name,
-                base_url=f"http://{ip}:{ports[0]}/fastapi",
-                browser_url=f"http://{ip}:{ports[0]}/steel-api"
-                f"/{runtime_token}",
-                front_browser_ws=f"ws://{ip}:"
-                f"{ports[0]}/steel-api/"
-                f"{runtime_token}/v1/sessions/cast",
-                client_browser_ws=f"ws://{ip}:"
-                f"{ports[0]}/steel-api/{runtime_token}/&sessionId"
-                f"={BROWSER_SESSION_ID}",
-                artifacts_sio=f"http://{ip}:{ports[0]}/v1",
-                ports=[ports[0]],
-                mount_dir=str(mount_dir),
-                storage_path=storage_path,
-                runtime_token=runtime_token,
-                version=image,
-            )
+            if self.container_deployment == "agentrun":
```

Comment:
We shouldn’t special-case agentrun here.
The logic for removing protocols or omitting ports should be encapsulated on the agent run client side.
This method should simply use a unified endpoint-building routine for all deployments, with the client responsible for adjusting the format as needed.
Doing this will keep the container creation code clean, avoid branching, and ensure deployment-specific differences are handled by their respective clients.

Reply:
OK, Agentrun Client was updated for adapting. Because sandbox deploying on Agentrun will provide a public HTTPS endpoint url, protocal variable was added.